crates/forge/src/collecting.rs

```diff
@@ -0,0 +1,588 @@
+use crate::{RunnerConfig, RunnerParams, BUILTINS};
+use anyhow::{anyhow, Context, Result};
+use assert_fs::fixture::{FileTouch, PathChild, PathCopy};
+use assert_fs::TempDir;
+use cairo_lang_sierra::program::Program;
+use camino::Utf8PathBuf;
+use rayon::prelude::{IntoParallelRefIterator, ParallelIterator};
+use test_collector::{collect_tests, LinkedLibrary, TestCase};
+use walkdir::WalkDir;
+
+pub struct TestsFromCrate {
+    pub sierra_program: Program,
+    pub test_cases: Vec<TestCase>,
+    pub test_crate_type: TestCrateType,
+}
+
+impl TestsFromCrate {
+    pub fn filter_by_name(&self, filter: &str, exact_match: bool) -> Self {
```

Comment:
This way it's easier to test the filtering behavior, because we have a specific functions that can be called in unit tests. Otherwise we would have to test by calling a higher level function.

Reply:
If you would implemented it idomatically with iterator you could possibly only test matching function.